### ISSUE TITLE
ZIM minorVersion of recent Zimit ZIM, 100rabbits_en_2024-01.zim, set to unknown value (2)

### ISSUE THREAD

**Issue by Jaifroid** [openzim/warc2zim#153]
Unexpectedly, the `minorVersion` field of the ZIM archive header of this ZIM is set to a value that is not declared in the [specification](https://wiki.openzim.org/wiki/ZIM_file_format), namely 2. This breaks the PWA, currently, due to detection of ZIM type in order to handle the different use of namespaces. Hitherto we only had values 0 and 1, and a value of 2 is not defined by the specification.

Is this intentional? What does a `minorVersion` of 2 signal to the client? How should I deal with this?

@mgautierfr, any clue?

**Comment by Jaifroid** [comment]
Thanks. Unfortunately (but hey, diversity is strength!), we cannot yet use libzim as our only backend in Kiwix JS and Kiwix PWA because the libzim WASM doesn't run, or runs very slowly, in too many of the contexts we actively support. Hence, we (or rather I) need to know what such changes mean and how we have to deal with them in our custom, pure JS backend.

I'd be grateful for any further info about what I need to support here, apart from accepting a value of 2, which currently throws an error "Unknown ZIM minor version!" in the backend of the PWA. (The Browser Extension is less affected, because it has fewer reading modes to support.)

**Comment by mgautierfr** [comment]
> I'd be grateful for any further info about what I need to support here

Nothing. It is just to say that the zim file may have alias (two entries pointing to the same clusterid/blobid). Nothing else has changed.

> which currently throws an error "Unknown ZIM minor version!" in the backend of the PWB.

You shouldn't refuse greater minor version. Minor version update are for compatible change. If your library is able to read `M.N` version, it will be able to read `M.N+1` (maybe not with all the features, but it should not break). You should stop on `M+1` however.

**Comment by Jaifroid** [comment]
OK, thanks -- I'll just change that test to a warning instead of an error. It is useful for the backend to warn that it is dealing with an unknown version. For us at least, the previous change of `minorVersion` from 0 to 1 (which completely changed the use of namespaces) wasn't a compatible change, and required quite extensive work in our backend, hence the caution around minorVersion changes.

By the way, I might be mistaken but this looks like a public version of a Zimit 2.0 ZIM. I need to test these ZIM types -- probably especially with regard to external link handling -- ideally before we start releasing them to the public. Do we now have a set of such ZIM archives for testing? I'm not sure if I need to inject Wombat into such archives in the reader, for example.

**Comment by kelson42** [comment]
> By the way, I might be mistaken but this looks like a public version of a Zimit 2.0 ZIM. I need to test these ZIM types -- probably especially with regard to external link handling -- ideally before we start releasing them to the public. Do we now have a set of such ZIM archives for testing? I'm not sure if I need to inject Wombat into such archives in the reader, for example.

Your are mixing ZIM format and Zimit 2.0. ZIM format has not changed beside the small thing explained by @mgautierfr. Zimit 2.0 has not been released yet like everybody knows. AFAIK Zimit 2.0 ZIM files should work out of the box in Kiwix. For now, its a bit early to test for you IMHO.

But the list of ZIM recipe files to assess is here https://github.com/openzim/zim-requests/issues/766#issuecomment-1890481096

**Comment by Jaifroid** [comment]
@kelson42  `100rabbits_en_2024-01.zim` is a Zimit 2.0 file, and it is in the public zimit directory on download.kiwix.org. Please see the recipe here: https://farm.openzim.org/recipes/100rabbits/config. It clearly says zimit2. If it is in a public download directory, it has (possibly mistakenly) been "released" and I need to test. It so happens that this is also `minorVersion` 2, and that `minorVersion` breaks the PWA, and I was given no warning about it. Of course I know that `minorVersion` 2 does not equate to Zimit 2, but this particular ZIM happens to be both at the same time.

**Comment by kelson42** [comment]
@Jaifroid This is an error.
@benoit74 Can you please fix it (move the file)?
@RavanJAltaie what happened here? Please ensure nothing is oublished using Zimit2 for the moment.

**Comment by benoit74** [comment]
I think this has most probably been done by @Popolechien. Probably because he was happy about the result. But I agree this is probably a bit too soon. I've probably failed to mention this is a Zimit2 file when I discussed the fact that description, title and filename should be updated.

I've moved the file to dev and fixed the recipe for now.

**Comment by Popolechien** [comment]
Yep my bad, I tested the file and was happy with the result, I didn't know (or did not listen when told) that zimit2 files should be held back for the time being. Sorry.

**Comment by benoit74** [comment]
No worries, I like what you fixed in the ZIM metadata, and this is not a big error :D (or a big thing to fix)

**Comment by Jaifroid** [comment]
Thanks, all 😊. I think we're now all on the same page, so I'm closing this issue as resolved. I've added support for `minorVersion > 1` in the PWA and Browser Extension (along with preliminary support for zimit2, though this is not yet released in the Browser Extension).

I suggest would should ensure all actively maintained readers are tested, patched if necessary with the required libkiwix update to support external link detection (?), and working before we start to release these in the production server. For the PWA, I've released an interim update supporting zimit2, and the Browser Extension should be ready for a new release sometime next week, probably in first week of Feb.
